### DISCUSSION TITLE
Spending Gitcoin grants funds

### DISCUSSION THREAD

**Discussion by ErikBjare** [uniswap-python/uniswap-python#181] [General]
The [Gitcoin grant](https://gitcoin.co/grants/2631/uniswap-python) has received a lot of love, and we want to use the funds to motivate/engage more developers on the project. 

Tasks that are up for grabs include: finishing v3 support (#95), adding new features (like #93, #160), more complex examples, improved documentation, etc.

The funds are here, and at time of writing has the following:
 - Mainnet (~0.2 ETH + ~$400): https://etherscan.io/address/0x0D8062911fBCC20bbC7B10f06D108d45Ad2E8c7F
 - zkSync (~0.8 ETH + ~$1800): https://zkscan.io/explorer/accounts/0x0d8062911fbcc20bbc7b10f06d108d45ad2e8c7f

So to sum up, we have roughly $7000 to spend (at current ETH prices).

I want to immediately nominate @liquid-8 to receive parts of the funds, as they've done a fair bit of work making PRs and being active in the issues. I'm open to suggestions regarding what would be reasonable (erring on the side of generous) compensation :)

As I've discussed with @shanefontaine before, we also intend on setting up a Gnosis Safe as a multisig, but I've been postponing this since I'm not sure how that works with zkSync.

**Comment by liquid-8** [comment]
Yo

**Reply by ErikBjare** [reply]
What do you think?

**Reply by liquid-8** [reply]
@ErikBjare Have a question btw. Do I need anything but account @ eth-mainnet for this?

**Reply by ErikBjare** [reply]
A mainnet address is all you need.

**Reply by ErikBjare** [reply]
I know I'm being a bit slow with this btw, will get around to disbursing next week.

**Reply by liquid-8** [reply]
Any news about this? Don't get me wrong, I'm just being curious.

**Reply by ErikBjare** [reply]
@liquid-8 Throw me an ETH address and I'll get around to it asap :)

**Reply by liquid-8** [reply]
Check your email, please. Sent from moyovedro@gmail.com, address 0xc207***61E8

**Reply by liquid-8** [reply]
Updated information has been sent from the same address.

**Comment by liquid-8** [comment]
Actually, current implementation itself misses not that much. There are things that you've mentioned above. If we assume using in production then I think it should be added options for the transaction replacement and more flexible gas price/gas volume manipulation too. Nevertheless, the most part of required functionality has been implemented already. So I'd leave some part of the fund (dunno how much exactly, 30%? 40?) as bounties for next PRs definitely. Concerning the rest of the funds I'm strongly sure it is up to you and Shane.

Also I would like to express fair gratitude for the nomination. Thanks my mom, God and Latin script. That is really pleasant and unexpected.

**Comment by shanefontaine** [comment]
I think this makes sense! Maybe 50% goes to @ErikBjare , 25% to @liquid-8 , and 25% for future PRs. If everyone agrees with that, then I'll work with Erik and get the Gitcoin claim/Gnosis Safe set up.

**Reply by liquid-8** [reply]
As I said it's up to you, guys. If Erik agreed then I'm totally fine with that.

**Reply by ErikBjare** [reply]
I finally got around to paying out to myself (should have done it in 2022 when the work was done). I found that there was ~$7962 in stables + UNI that I sold for USDC, somewhat short of the $10,000 in my allocation from the Uniswap Grant (iirc, can only find [Adams](https://twitter.com/adamscochran/status/1555243590250647552) tweet reference to the $20,000). This is completely my bad though, should have paid out immediately.

Txs:
 - $2787 USDC
   - https://zkscan.io/explorer/transactions/0x0f49ec68afb5d86e8b9c03f57b58fa93fcc8167b7d9efbc64c755a003ee83b57
 - $5175 DAI
   - https://etherscan.io/tx/0x97bf73e66669b8fcde1ddddccfdfdd6effbca83604a92e7da3dbd0446b7cba0d

For a total of $7962.

**Comment by ErikBjare** [comment]
Just paid out ~$1800 to @liquid-8 from the zkSync wallet. Some of it in DAI, and some in ETH (to minimize fees, since the amount of USDC and USDT in the wallet were only ~$200 each, with a ~$30 fee per tx).

Transactions:
 - 1323 DAI: https://zkscan.io/explorer/transactions/0xbef0d9560a261176a462191fc20c3b6379de54fa87ded2d433cfe4d3018ceefe
 - 0.1 ETH (~$460): https://zkscan.io/explorer/transactions/0xbb9a32f95f02ad193f5f2767ec69bfa9829781729ce8a3c452d885bd13fea081

Funds should arrive between 30min and 7h, according to zkSync.

**Reply by liquid-8** [reply]
Thanks a lot!

**Reply by ErikBjare** [reply]
Thank you for all the work! 🤗

**Comment by invnn** [comment]
Is there full polygon network support?

After some testing I've noticed some features work and others dont.
(some pool info/wallet balances can be pulled. Making swaps doesnt seem to work)

**Comment by t270811** [comment]
Thanks

**Comment by Crypton-56** [comment]
Hello! I spent a lot of time on Gitcoin Passport
